File: Core/Entities/OrderAggregate/Order.cs
using Core.Interfaces;

namespace Core.Entities.OrderAggregate;

public class Order : BaseEntity, IDtoConvertible
{
    public DateTime OrderDate { get; set; }
    public OrderStatus Status { get; set; }
    public required string BuyerEmail { get; set; }
    public required ShippingAddress ShippingAddress { get; set; }
    public required PaymentSummary PaymentSummary { get; set; }
    
    public required DeliveryMethod DeliveryMethod { get; set; }
    public required List<OrderItem> OrderItems { get; set; }
    public decimal Subtotal { get; set; }
    public required string PaymentIntentId { get; set; }

    public decimal GetTotal()
    {
        return Subtotal + DeliveryMethod.Price;
    }
}

File: client/src/app/shared/modules/order.ts
export type Order = {
    id: number
    orderDate: string
    orderStatus: string
    buyerEmail: string
    shippingAddress: ShippingAddress
    shippingPrice: number
    deliveryMethod: string
    paymentSummary: PaymentSummary
    orderItems: OrderItem[]
    subtotal: number
    total: number
  }
  
  export type ShippingAddress = {
    name: string
    line1: string
    line2: any
    city: string
    state: string
    postalCode: string
    country: string
  }
  
  export type PaymentSummary = {
    last4: number
    brand: string
    expMonth: number
    expYear: number
  }
  
  export type OrderItem = {
    id: number
    productName: string
    pictureUrl: string
    price: number
    quantity: number
  }

  export type OrderToCrete = {
    cartId: string
    shippingAddress: ShippingAddress
    deliveryMethodId?: number
    paymentSummary: PaymentSummary
  }
  